### ISSUE TITLE
Shipping method name missing

### ISSUE THREAD

**Issue by GoTeamScotch** [academe/wc-multiple-packages#10]
When viewing Admin > WooCommerce > Settings > Shipping (/wp-admin/admin.php?page=wc-settings&tab=shipping), the name for the shipping method is absent.

![Example](http://i.imgur.com/qumVGoL.jpg)

Environment:
WooCommerce v2.4.5
WordPress v4.3
PHP v5.6.11
MySQL v5.6.25 
Apache v2.4.16

**Comment by judgej** [comment]
I'm not entirely sure it is possible to give this a name, since it is not a shipping method, but a shipping method modifier. I'll have another look - and if you know of a way to do this, or know of any other plugins that provide package grouping and _do_ show a name here, I will be happy to take a look and fix it.

It appears in this list as a consequence of being added to the top links in the shipping methods page, and it is the only way to get to the admin page for it. An alternative approach would be to take it off this table (and the top links) completely and provide an independent link to the admin page for this, though that will bring a lot of its own coding overhead.

Let me know if you see a way around this. I could just be missing something obvious.

**Comment by judgej** [comment]
Some themes hide the top links and only display this table, but with "edit" links on each row.

**Comment by GoTeamScotch** [comment]
Found it. WooCommerce's WC_Shipping_Method class has both "method_title" AND "title" properties which you can define in the constructor.

So in \classes\Academe_Multiple_Packages_Settings.php...

Instead of

```
$this->method_title = __('Packages Grouping', 'academe-package-config-woo');
```

use

```
$this->method_title = $this->title = __('Packages Grouping', 'academe-package-config-woo');
```

(or whatever name you want to give it)

I did some poking around trying to find where else "title" is used (as opposed to "method_title") and the only place I saw it was on the settings page in the picture I attached previously. It doesn't appear on the front-end.

Unrelated - I noticed in the Packages Grouping settings tab that you deselect the Multiple Packages shipping method for a given shipping class. Is that intentional? If not you can just remove it from the array where output_shipping_class_method_restrictions() is defined.

```
unset($shipping_methods['academe_multiple_packages']);
```

(just before $total_shipping_methods)

**Comment by judgej** [comment]
You're a star - I'll get that fixed ASAP.

That unset doesn't look right. I wonder if it got left in by accident while developing/testing? Can't remember - will look at that.

**Comment by GoTeamScotch** [comment]
The unset is my addition. It's just an easy way to remove it from being shown on the shipping class restriction table. There's a separate hook in WooCommerce that you could use instead but this way works and winds up being less code.

**Comment by judgej** [comment]
I'm a little unclear about where the unset thing applies, so I'll make a bugfix release for the main topic of this issue, but keep it open until I understand. There are various other refactoring tasks I need to do in this plugin, so I'll pull the unset into that.

**Comment by judgej** [comment]
Fix released as 1.2.2 on WordPress.org

Strangely, I did not need to tag the release in SVN - I just uploaded the new code and wp.org picked it up as a new version immediately. I thought it waited to release the new version until you added a tag to the SVN repository, but apparently not.
